unix-opts.lisp

```diff
@@ -514,15 +539,17 @@ text is wider than ARGUMENT-BLOCK-WIDTH."
                                               (length (car el)))
                                             option-strings)
                                     :initial-value 0)))
-      (loop
-        :for (opt-meta . opt-description) :in option-strings
-        :for newline = (>= (length opt-meta)
-                           argument-block-width)
-        :do (format stream "  ~a~a~%"
-                    (pad-right opt-meta (+ (if newline 0 1) max-opts-length))
-                    (add-text-padding opt-description
-                                      :padding (+ 3 max-opts-length)
-                                      :newline newline)))
+      (let ((padding (+ 3 max-opts-length)))
+        (loop
+          :for (opt-meta . opt-description) :in option-strings
+          :for newline = (>= (length opt-meta)
+                             argument-block-width)
+          :do (format stream "  ~a~a~%"
+                      (pad-right opt-meta (+ (if newline 0 1) max-opts-length))
+                      (add-text-padding opt-description
+                                        :padding padding
+                                        :newline newline
+                                        :max-width (- max-width padding)))))
```

Comment:
I would prefer to call `add-text-padding` with the actual `max-width` value and let it handle subtractions for the padding.